feat: W-16857079 - add a --snapshot flag for creating scratch org (#1358)
salesforcecli/plugin-org: messages/create_scratch.md

# description

There are four ways to create a scratch org:

    * Specify a definition file that contains the scratch org options.
    * Use the --edition flag to specify the one required option; this method doesn't require a defintion file.

>     * Use the --snapshot flag to create a scratch org from a snapshot. Snapshots are a point-in-time copy of a scratch org that you create with the "sf org create snapshot" command.

Reviewer: ```suggestion
    * Use the --snapshot flag to create a scratch org from a snapshot. Snapshots are a point-in-time copy of a scratch org; you create a snapshot with the "sf org create snapshot" command.
```

# flags.source-org.summary

> 15-character ID of the org shape that the new scratch org will be based on. Overrides the value of the "sourceOrg" option in the definition file, if set.

Reviewer: ```suggestion
15-character ID of the org shape that the new scratch org is based on. Overrides the value of the "sourceOrg" option in the definition file, if set.
```
